src/main/kotlin/app/SourcererApi.kt

```diff
@@ -25,74 +26,69 @@ object SourcererApi {
     val password
         get() = Configurator.getPassword()
 
-    @Throws(IOException::class, HttpException::class)
-    fun getUserBlocking(): User {
-        val name = "getUserBlocking"
-        Logger.debug("Request $name initialized")
-
-        val (request, response, result) = Fuel.get("/user/info")
-                .authenticate(username, password)
-                .responseString()
-        val body = result.get()
-        Logger.debug("Request $name success")
-        return User().parseFrom(body)
+    private fun createRequestGetUser(): Request {
+        return Fuel.get("/user/info").authenticate(username, password)
     }
 
-    @Throws(IOException::class, HttpException::class)
-    fun postCommitBlocking(commit: Commit): String {
-        val name = "postCommitBlocking"
-        Logger.debug("Request $name initialized")
-        val (request, response, result) = Fuel.get("/commit")
-                .authenticate(username, password)
+    private fun createRequestPostCommit(commit: Commit): Request {
+        return Fuel.post("/commit").authenticate(username, password)
                 .body(commit.serialize())
-                .responseString()
+    }
+
+    private fun <T> makeBlockingRequest(request: Request, requestName: String,
+                                parser: (String) -> T): T {
```

Comment:
I would recommend to align arguments under each other. Will be more readable. E.g.
```
private fun<T> makeBlockingRequest(request: Request, requestName: String,
                                   parser: (String) -> T): T {
```